compiler/optimizer/InductionVariable.cpp

```diff
@@ -4724,6 +4732,110 @@ void TR_LoopStrider::replaceLoadsInSubtree(
       node->setAndIncChild(0, lload);
       exLoads.add(node);
       }
+
+   widenComparison(node, iv, defRHS, exLoads);
+   }
+
+/**
+ * Widen comparisons against \p iv.
+ *
+ * Signed 32-bit comparisons made directly against \p iv, or directly against
+ * the node that produces the new value of \p iv, are changed into the
+ * corresponding 64-bit comparisons, by wrapping each child in \c i2l.
+ *
+ * For example, when replacing the 32-bit \c i with the 64-bit \c k, consider
+ * the following comparison:
+ *
+   \verbatim
+      ificmpge
+        iload i
+        iload n
+   \endverbatim
+ *
+ * In the course of replaceInStructure(TR_Structure*), it will be rewritten as:
+ *
+   \verbatim
+      ificmpge
+        l2i
+          lload k
+        iload n
+   \endverbatim
+ *
+ * In order to make it  into a more obvious test of \c k, we change it into a
+ * 64-bit comparison using \c i2l like so:
+ *
+   \verbatim
+      iflcmpge
+        i2l
+          l2i
+            lload k
+        i2l
+          iload n
+   \endverbatim
+ *
+ * Because the \c l2i node here is an ex-load, it will cancel the \c i2l later:
```

Comment:
Okay thanks for the explanation; it makes sense since it was not a random lload but one created by the strider for an ind var that was an int originally.